core/src/main/java/io/grpc/Context.java

```diff
@@ -261,12 +249,11 @@ public CancellableContext withDeadlineNanoTime(long deadlineNanoTime,
    *   });
    * </pre>
    */
-  public CancellableContext withDeadlineAfter(long duration, TimeUnit unit,
+  public CancellableContext withDeadline(Deadline deadline,
```

Comment:
Could we keep `withDeadlineAfter` as a convenience? I didn't think anyone had a problem with it, and it was the one I was expecting to be used most by far.